packages/babel-core/src/transformation/normalize-opts.ts

```diff
@@ -30,7 +39,10 @@ export default function normalizeOptions(config: ResolvedConfig): {} {
       sourceType:
         path.extname(filenameRelative) === ".mjs" ? "module" : sourceType,
 
-      sourceFileName: filename,
```

Comment:
The plugin `babel-plugin-bundled-import-meta` is relying on this option to get the complete input filename, 
https://github.com/cfware/babel-plugin-bundled-import-meta/blob/master/index.js#L67

We reverted the change in https://github.com/babel/babel/pull/13732, we will probably have to defer this PR to Babel 8.